src/maxInlineDeclarationsRule.ts

```diff
@@ -1,110 +1,130 @@
 
 export class Rule extends Lint.Rules.AbstractRule {
-  public static metadata: Lint.IRuleMetadata = {
-    ruleName: 'max-inline-declarations',
-    type: 'maintainability',
-    description: 'Disallows having too many lines in inline template or styles. Forces separate template or styles file creation.',
+  static metadata: Lint.IRuleMetadata = {
+    description: 'Disallows having too many lines in inline animations, styles or template. Forces separate file creation.',
     descriptionDetails: 'See more at https://angular.io/guide/styleguide#style-05-04',
+    hasFix: false,
+    optionExamples: ['[{animations: 20, styles: 8, template: 5}]'],
     options: {
-      type: 'array',
       items: {
-        type: 'object',
-      }
+        type: 'object'
+      },
+      type: 'array'
     },
-    optionsDescription: 'Define inline template and styles lines limit.',
-    optionExamples: ['[{template: 5, styles: 8}]'],
-    typescriptOnly: true,
-    hasFix: false
+    optionsDescription: 'Define inline animations, styles and template lines limit.',
+    ruleName: 'max-inline-declarations',
+    type: 'maintainability',
+    typescriptOnly: true
   };
 
-  private readonly templateLinesLimit: number = 3;
-  private readonly stylesLinesLimit: number = 3;
-
-  constructor(options: IOptions) {
-    super(options);
-    if (options.ruleArguments.length > 1) {
-      const args = options.ruleArguments[1];
-      if (args.template > -1) {
-        this.templateLinesLimit = args.template;
-      }
-      if (args.styles > -1) {
-        this.stylesLinesLimit = args.styles;
-      }
-    }
-  }
+  static FAILURE_STRING = 'Exceeds maximum allowed inline %s (%s). Consider moving this to a separate file.';
 
-  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-    return this.applyWithWalker(new MaxInlineDeclarationsValidator(sourceFile, this, this.templateLinesLimit, this.stylesLinesLimit));
+  apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    return this.applyWithWalker(new MaxInlineDeclarationsWalker(sourceFile, this.getOptions()));
   }
 }
 
-export class MaxInlineDeclarationsValidator extends NgWalker {
+export type MetadataType = 'animations' | 'styles' | 'template';
 
-  private newLineRegExp = /\r\n|\r|\n/;
+const generateFailure = (type: MetadataType, limit: number): string => {
+  return sprintf.apply(null, [Rule.FAILURE_STRING, type, limit]);
+};
 
-  constructor(sourceFile: ts.SourceFile, private rule: Rule, private templateLinesLimit: number, private stylesLinesLimit: number) {
-    super(sourceFile, rule.getOptions());
+export const animationsFailure = (limit: number): string => {
+  return generateFailure('animations', limit);
+};
+
+export const stylesFailure = (limit: number): string => {
+  return generateFailure('styles', limit);
+};
+
+export const templateFailure = (limit: number): string => {
+  return generateFailure('template', limit);
+};
+
+export const animationsLinesDefaultLimit = 7;
```

Comment:
Not sure about this default limit...

Reply:
It seems to me too little. Animation is quickly verbose. 15 lines, it is more common for a simple animation.